// angular-calendar-scheduler/esm2020/modules/scheduler/utils/calendar-scheduler-utils.provider.mjs
import { Injectable } from '@angular/core';
import { getSchedulerViewHourGrid, getSchedulerViewDays, getSchedulerView } from './calendar-scheduler-utils';
import * as i0 from "@angular/core";
import * as i1 from "angular-calendar";
export class CalendarSchedulerUtils {
    constructor(dateAdapter) {
        this.dateAdapter = dateAdapter;
    }
    getSchedulerViewHourGrid(args) {
        return getSchedulerViewHourGrid(this.dateAdapter, args);
    }
    getSchedulerViewDays(args) {
        return getSchedulerViewDays(this.dateAdapter, args);
    }
    getSchedulerView(args) {
        return getSchedulerView(this.dateAdapter, args);
    }
}
CalendarSchedulerUtils.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: CalendarSchedulerUtils, deps: [{ token: i1.DateAdapter }], target: i0.ɵɵFactoryTarget.Injectable });
CalendarSchedulerUtils.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: CalendarSchedulerUtils });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: CalendarSchedulerUtils, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.DateAdapter }]; } });
