// angular-calendar-scheduler/esm2020/modules/scheduler/scheduler-config.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Auth configuration.
 */
export class SchedulerConfig {
    constructor(config = {}) {
        this.locale = 'en';
        this.headerDateFormat = 'daysRange';
        this.logEnabled = false;
        function use(source, defaultValue) {
            return config && source !== undefined ? source : defaultValue;
        }
        this.locale = use(config.locale, this.locale);
        this.headerDateFormat = use(config.headerDateFormat, this.headerDateFormat);
        this.logEnabled = use(config.logEnabled, this.logEnabled);
    }
}
SchedulerConfig.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: SchedulerConfig, deps: [{ token: SchedulerConfig }], target: i0.ɵɵFactoryTarget.Injectable });
SchedulerConfig.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: SchedulerConfig });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: SchedulerConfig, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: SchedulerConfig }]; } });
