// angular-calendar-scheduler/esm2020/modules/scheduler/pipes/scheduler-event-title.pipe.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../formatters/scheduler-event-title-formatter.provider";
export class SchedulerEventTitlePipe {
    constructor(schedulerEventTitle) {
        this.schedulerEventTitle = schedulerEventTitle;
    }
    transform(title, titleType, event) {
        return this.schedulerEventTitle[titleType](event);
    }
}
SchedulerEventTitlePipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: SchedulerEventTitlePipe, deps: [{ token: i1.SchedulerEventTitleFormatter }], target: i0.ɵɵFactoryTarget.Pipe });
SchedulerEventTitlePipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "14.1.1", ngImport: i0, type: SchedulerEventTitlePipe, name: "schedulerEventTitle" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: SchedulerEventTitlePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'schedulerEventTitle'
                }]
        }], ctorParameters: function () { return [{ type: i1.SchedulerEventTitleFormatter }]; } });
